cfft/src/radix_2_butterfly.rs

```diff
@@ -0,0 +1,278 @@
+use alloc::vec::Vec;
+
+use p3_field::{ComplexExtension, Field};
+use p3_matrix::dense::RowMajorMatrix;
+use p3_matrix::Matrix;
+
+use crate::traits::CircleSubgroupFFT;
+use crate::util::{cfft_domain, twin_coset_domain};
+
+/// An O(N Log(N)) implementation of the CFT roughly analagous to the FFT approach to the DFT.
+/// See the paper "Circle Stark".
+#[derive(Default, Clone)]
+pub struct Radix2CFT;
+
+impl<Base: Field, Ext: ComplexExtension<Base>> CircleSubgroupFFT<Base, Ext> for Radix2CFT {
+    type Evaluations = RowMajorMatrix<Base>;
+
+    fn cfft_batch(
+        &self,
+        mut mat: RowMajorMatrix<Base>,
+    ) -> <Self as CircleSubgroupFFT<Base, Ext>>::Evaluations {
+        let n = mat.height();
+        let n_u32: u32 = n.try_into().unwrap();
+        let log_n: usize = n.trailing_zeros().try_into().unwrap();
+        debug_assert_eq!(1_u32 << log_n, n_u32); // The height better be a power of 2.
+
+        let width = mat.width();
+
+        let twiddles = cfft_twiddles::<Base, Ext>(log_n); // These should be precomputed.
+
+        for (i, twiddle) in twiddles.iter().enumerate() {
+            let block_size = 1 << (log_n - i);
+            let half_block_size = block_size >> 1;
+
+            for chuncks in mat.values.chunks_mut(block_size * width) {
+                let (low_chunks, high_chunks) = chuncks.split_at_mut(half_block_size * width);
+
+                low_chunks
```

Comment:
Should use `itertools::izip!`. Normally this would be bikeshedding but it actually helps with performance

Reply:
also `chunks_exact_mut`